app/src/main/kotlin/at/bitfire/davdroid/ui/AppSettingsScreen.kt

```diff
@@ -492,15 +506,118 @@ fun AppSettings_Integration(
     )
 
     val context = LocalContext.current
+    val pm = context.applicationContext.packageManager
+
+    var distributor by remember { mutableStateOf(UnifiedPush.getSavedDistributor(context)) }
+    val distributors = remember { UnifiedPush.getDistributors(context) }
+    LaunchedEffect(Unit) {
+        UnifiedPush.getAckDistributor(context)?.let {
+            UnifiedPush.registerApp(context)
```

Comment:
This is an action that affects more than the UI, so it should also be done by the model.